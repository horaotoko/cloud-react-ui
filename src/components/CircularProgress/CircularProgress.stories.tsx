import { Story, Meta } from '@storybook/react';
import { CircularProgress, Props } from './CircularProgress';

export default {
  component: CircularProgress,
  title: 'components/CircularProgress',
} as Meta;

const Template: Story<Props> = (args) => <CircularProgress {...args} />;

export const Default = Template.bind({});
