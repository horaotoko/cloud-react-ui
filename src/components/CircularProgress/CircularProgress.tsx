import { forwardRef } from 'react';
import styled, { css } from 'styled-components';
import { CircularProgress as MuiCircularProgress } from '@material-ui/core';
import { defaultProps } from '../../theme';

export type Props = {
  className?: string;
  size?: number | string;
};

const StyledCircularProgress = styled(MuiCircularProgress)`
  ${({ theme: { circularProgress } }) => css`
    .MuiCircularProgress-circle {
      color: ${circularProgress.color};
    }
  `}
`;
StyledCircularProgress.defaultProps = defaultProps;

const CircularProgress = forwardRef<HTMLDivElement, Props>((props, ref) => (
  <StyledCircularProgress ref={ref} {...props} />
));

CircularProgress.displayName = 'CircularProgress';

export { CircularProgress };
