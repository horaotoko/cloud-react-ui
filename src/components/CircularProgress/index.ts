export { CircularProgress } from "./CircularProgress";
export type { Props as CircularProgressProps } from "./CircularProgress";
