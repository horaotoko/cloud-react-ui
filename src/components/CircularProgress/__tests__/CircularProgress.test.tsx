import { render } from "@testing-library/react";
import { CircularProgress } from "../CircularProgress";

describe("CircularProgress", () => {
  it("snapshot test", () => {
    const { asFragment } = render(<CircularProgress />);
    expect(asFragment()).toMatchSnapshot();
  });
});
