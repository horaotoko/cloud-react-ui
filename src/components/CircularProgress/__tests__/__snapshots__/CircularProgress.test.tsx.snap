// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularProgress snapshot test 1`] = `
<DocumentFragment>
  .c0 .MuiCircularProgress-circle {
  color: #3B7DE9;
}

<div
    class="MuiCircularProgress-root c0 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style="width: 40px; height: 40px;"
    theme="[object Object]"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</DocumentFragment>
`;
